docs/src/references/c8y-configuration-management.md

```diff
@@ -163,3 +279,203 @@ Note that:
+one might have to install a protocol adaptor on the main device).
+
+Here is the protocol that has to be implemented by the child-device configuration client.
+This protocol covers 4 interactions, the child devices:
+1. Connecting to thin-edge
+2. Downloading configuration file updates from thin-edge
+3. Uploading current configuration files to thin-edge
+4. Notifying the list of configuration files to thin-edge
+
+### The child device connects to the thin-edge parent device
+
+From a TCP point of view, the child devices act as clients
+and all the connections to thin-edge are established by the child devices.
+* Thin-edge opens two ports for MQTT and HTTP over the local network.
+  These ports are controlled on the main device `tedge config`:
+  * `mqtt.external.bind_address`
+  * `mqtt.external.port`
+  * `http.external.port`
+* The child devices must know the main device IP address.
+  * This is the address set for `mqtt.external.bind_address` on the main device.
+  * For the very specific case, where the child-device agent runs on the main device,
+    this connection address can be the `localhost`.
+* On start, the child-device agent for configuration management,
+  must connect over MQTT to thin-edge device.
+  * It must use a session name that is unique on the thin-edge bus.
+    `"$CHILD_DEVICE_ID/configuration"` is the recommendation.
+  * It must subscribe to `tedge/$CHILD_DEVICE_ID/commands/req/#` to receive requests.
+  * It has to publish responses under `tedge/$CHILD_DEVICE_ID/commands/res/#`.
+* On demand, the child-device agent has to send HTTP requests to thin-edge.
+  * These requests are REST requests (GET/PUT/DELETE) to exchange files.
+  * The urls are forged by thin-edge telling the child devices
+    where to get and put configuration files.
+* Currently, all these connections are done without TLS support.
+  * Hence, the main device must be configured so these MQTT and HTTP ports
+    are only open on a local trusted network.
+
+### The child device downloads configuration updates on notification
+
+When a configuration file for a child device
+is received from the cloud by the `c8y_configuration_plugin` daemon,
+the configuration plugin manages the transfer of this file to the child device
+and notifies the cloud on the progress of this configuration update operation.
+
+1. On reception of a configuration update for a child device named `$CHILD_DEVICE_ID`
+   of a file `{ path = $PATH, type = $TYPE }` defined in `$CHILD_DEVICE_ID/c8y-configuration-plugin.toml` 
+   the `c8y_configuration_plugin` downloads the new configuration content to the temporary directory defined by `tedge config get tmp.path`,
+   and, on success, moves this content under the thin-edge file transfer HTTP root.
+   * The file is moved to `$TEDGE_HTTP_ROOT/$CHILD_DEVICE_ID/config_update/$TYPE`
+     making this file available under `http://$TEDGE_HTTP/tedge/$CHILD_DEVICE_ID/config_update/$TYPE`.
```

Comment:
According tp the [file transfer specifications](https://github.com/thin-edge/thin-edge.io/pull/1361)

```suggestion
     making this file available under `http://$TEDGE_HTTP/tedge/file-transfer/$CHILD_DEVICE_ID/config_update/$TYPE`.
```